shared/src/akri/configuration.rs

```diff
@@ -333,118 +200,47 @@ mod crd_serializeation_tests {
     fn test_config_defaults_with_serialization() {
         let _ = env_logger::builder().is_test(true).try_init();
 
+        if serde_json::from_str::<Configuration>(r#"{}"#).is_ok() {
+            panic!("discovery handler is required");
+        }
+
+        serde_json::from_str::<Configuration>(
+            r#"{"discoveryHandler":{"name":"random", "discoveryDetails":{"key":"random protocol"}}}"#,
+        )
+        .unwrap();
+        if serde_json::from_str::<Configuration>(r#"{"discoveryHandler":{"name":"random"}}"#)
+            .is_err()
         {
-            if serde_json::from_str::<Configuration>(r#"{}"#).is_ok() {
-                panic!("protocol is required");
-            }
-            if serde_json::from_str::<Configuration>(r#"{"protocol":{"onvif":{}}}"#).is_err() {
-                panic!("onvif protocol doesn't require anything");
-            }
-            if serde_json::from_str::<Configuration>(r#"{"protocol":{"udev":{}}}"#).is_ok() {
-                panic!("udev protocol requires udevRules");
-            }
-            if serde_json::from_str::<Configuration>(r#"{"protocol":{"opcua":{}}}"#).is_ok() {
-                panic!("opcua protocol requires one opcua discovery method");
-            }
+            panic!("discovery details are not required");
+        }
+        if serde_json::from_str::<Configuration>(r#"{"discoveryHandler":{}}"#).is_ok() {
+            panic!("discovery handler name is required");
         }
 
-        let json = r#"{"protocol":{"onvif":{}}}"#;
+        let json = r#"{"discoveryHandler":{"name":"onvif", "discoveryDetails":{"discoveryHandlerConfig":"{\"onvif\":{}}"}}}"#;
```

Comment:
is discoveryHandlerConfig (or the second onvif) required?

is this valid `r#"{"discoveryHandler":{"name":"onvif", "discoveryDetails":{"discoveryHandlerConfig":"{}"}}}"#;`?

or this: `r#"{"discoveryHandler":{"name":"onvif", "discoveryDetails":{}}}"#;`?

Reply:
`r#"{"discoveryHandler":{"name":"onvif", "discoveryDetails":{}}}"#;` is sufficient. This is just testing that you can have serialized yaml or json as a value of the `discovery_details` map, like we do for our supported protocols

Reply:
switched to string instead of map `discovery_details` in 8698b3197b9